File: oms/src/main/java/com/vjentrps/oms/service/ReportsService.java
package com.vjentrps.oms.service;

import java.util.List;

import com.vjentrps.oms.exception.OmsServiceException;
import com.vjentrps.oms.model.ProductStock;
import com.vjentrps.oms.model.SearchObj;
import com.vjentrps.oms.model.StockRecord;

public interface ReportsService {
	 
	 List<ProductStock> fetchProductStock() throws OmsServiceException;
	 
	 List<StockRecord> fetchStockRecords() throws OmsServiceException;

     List<StockRecord> fetchStockRecords(SearchObj searchObj) throws OmsServiceException, Exception;

	 ProductStock fetchProductStock(long productId) throws OmsServiceException;  
	
}

File: oms/src/main/webapp/resources/js/service/reports_service.js
'use strict';

omsApp.factory('ReportsService', ['$http', '$q', function($http, $q){

	return {

		fetchAllProductsStock: function() {
			return $http.get('http://ntiyyagura:8080/oms/service/reports/productStock')
			.then(
					function(response){
						return response.data.object;
					}, 
					function(errResponse){
						console.error('Error while fetching products stock');
						return $q.reject(errResponse);
					}
			);
		},
		fetchProductStock: function(productId) {
			return $http.get('http://ntiyyagura:8080/oms/service/reports/productStock/'+productId)
			.then(
					function(response){
						return response.data.object;
					}, 
					function(errResponse){
						console.error('Error while fetching productstock');
						return $q.reject(errResponse);
					}
			);
		},
		fetchAllStockRecords: function() {
			return $http.get('http://ntiyyagura:8080/oms/service/reports/stockRecord')
			.then(
					function(response){
						return response.data.object;
					}, 
					function(errResponse){
						console.error('Error while fetching stock records');
						return $q.reject(errResponse);
					}
			);
		},
		
		fetchStockRecords: function(searchObj) {
			return $http.post('http://ntiyyagura:8080/oms/service/reports/stockRecord/search', searchObj)
			.then(
					function(response){
						return response.data.object;
					}, 
					function(errResponse){
						console.error('Error while fetching stock records');
						return $q.reject(errResponse);
					}
			);
		}
		
		
	};

}]);
